recipes/caf/all/conanfile.py

```diff
@@ -0,0 +1,142 @@
+            },
+        }
+        return standards.get(cppstd) or {}
+
+    def validate(self):
+        cppstd = "11" if Version(self.version) <= "0.17.6" else "17"
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, cppstd)
+        min_version = self._minimum_compilers_version(cppstd).get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, cppstd, self.settings.compiler, self.settings.compiler.version))
+
+        if self.settings.compiler == "apple-clang" and Version(self.settings.compiler.version) > "10.0" and \
+                self.settings.arch == 'x86':
+            raise ConanInvalidConfiguration("clang >= 11.0 does not support x86")
+        if self.options.shared and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Shared libraries are not supported on Windows")
+        if self.options.with_openssl and self.settings.os == "Windows" and self.settings.arch == "x86":
+            raise ConanInvalidConfiguration("OpenSSL is not supported for Windows x86")
+
+    def _cmake_configure(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            if Version(self.version) <= "0.17.6":
+                self._cmake.definitions["CMAKE_CXX_STANDARD"] = "11"
+                self._cmake.definitions["CAF_NO_AUTO_LIBCPP"] = True
+                self._cmake.definitions["CAF_NO_OPENSSL"] = not self.options.with_openssl
+                for define in ["CAF_NO_EXAMPLES", "CAF_NO_TOOLS", "CAF_NO_UNIT_TESTS", "CAF_NO_PYTHON"]:
+                    self._cmake.definitions[define] = "ON"
+                self._cmake.definitions["CAF_BUILD_STATIC"] = not self.options.shared
+                self._cmake.definitions["CAF_BUILD_STATIC_ONLY"] = not self.options.shared
+            else:
+                self._cmake.definitions["CMAKE_CXX_STANDARD"] = "17"
+                self._cmake.definitions["CAF_ENABLE_OPENSSL_MODULE"] = self.options.with_openssl
+                for define in ["CAF_ENABLE_EXAMPLES", "CAF_ENABLE_TOOLS", "CAF_ENABLE_TESTING"]:
+                    self._cmake.definitions[define] = "OFF"
+            if tools.os_info.is_macos and self.settings.arch == "x86":
+                self._cmake.definitions["CMAKE_OSX_ARCHITECTURES"] = "i386"
+            self._cmake.definitions["CAF_LOG_LEVEL"] = self.options.log_level
+            if self.settings.os == 'Windows':
+                self._cmake.definitions["OPENSSL_USE_STATIC_LIBS"] = True
+                self._cmake.definitions["OPENSSL_MSVC_STATIC_RT"] = True
```

Comment:
Useless, those variables come from built-in find OpenSSL https://cmake.org/cmake/help/v3.7/module/FindOpenSSL.html
openssl recipe don't emulate them and there is not need to emulate them, because static/shared openss and static/shared runtimes are controlled by conan profile.

Reply:
This was in the original Bincrafters recipe.   It translates into compiler definitions used by the OpenSSL header files.

Reply:
I'll test removing it and see if it works.

Reply:
It works.